### PR TITLE
CocoaPods 1.6.0 beta blog post

### PR DESCRIPTION
**Pull request by dnkoutso**

### CODE REVIEW

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+From a user point of view, this should have zero effects when you compile your code. If there is anything that does not look right please file an [issue](https://github.com/CocoaPods/CocoaPods/issues/new) with us providing steps and a sample app showing the problem.
+
+#### LinkedIn Performance Case Study?
+
+TODO MAYBE INCLUDE THIS HERE
+
+### Better Test Spec Integration
+
+Test specs are becoming a fundamental piece of CocoaPods for pod authors. The ability to specify you test sources and have them execute through `lint` becomes an invaluable tool for making your `podspec` file source of truth for your pod. With 1.6.0, specifying multiple `test_spec` entries in your `podspec` will no longer merge them together under a single test bundle target.
```
Comment (orta) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: > Test specs are becoming a fundamental piece of CocoaPods for pod authors. The ability to specify you*r* test sources and have them execute through `pod lint` lets the feature become an invaluable tool for using the `podspec` as a source of truth for your library. With 1.6.0, specifying multiple `test_spec` entries in your `podspec` will no longer merge them together under a single test bundle target but instead let you create separate test suites.

Feels a bit better

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+_CocoaPods 1.6.0_ next release will focus on stability, performance and scalability for large projects through a longer beta cycle.
+
+<!-- more -->
+
+We are happy to announce that the first beta of 1.6.0 is out! For this release, we are revamping the build settings generation to be faster and easier to maintain. We've also improved integration of test specs for pods that have multple ones.
+
+### Revamped Build Settings
+
+The generation of build settings has been completely re-written. The primary goals for this re-write was to clean up the code but also allow this process to scale better on large projects. Having to apply a fix would often cause a cascading amount of performance regressions making it very hard to test but also apply the fix for everyone.
```
Comment (orta) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: > Having to apply a fix would often...

Probably worth stating who would be the applying the fixes

> When contributors to CocoaPods make build setting changes it would often cause a cascading amount of performance regressions. This makes it very hard to test but also to be sure the fix applies for everyone.

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+And now it looks like this:
+
+<center>
+{% breaking_image /assets/blog_img/CocoaPods-1.6.0-beta/test_specs_after.png %}
+</center>
+
+Much better!
+
+We _strongly_ support the idea of having tests for your code and aim to give the tools to support you doing so. Having your `.podspec` file be the source of truth and coupled with recently released [cocoapods-generate](https://github.com/square/cocoapods-generate) plugin your developer experience can be much improved.
```
Comment (orta) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: This is great 👍 - good show at pushing the community towards higher standards

Perhaps "and aim to give the tools to support you doing so" could be more aspirational?

- "and aim to provide the right tools for doing this simply"

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+```sh
+$ gem install cocoapods
+```
+
+Before we go, we always like to use this last bit of the blog post as an opportunity to express what we have in mind for the next release (currently aimed to be 1.7.0). More particularly, we would like to expand the `swift_version` DSL to support a range of Swift versions instead of just a single one and also 
+
+Thank you for making CocoaPods 1.6.0 a reality! We've come a long way.
+
+Checkout the [changelog](https://github.com/CocoaPods/CocoaPods/releases/tag/1.6.0) to get the full list of changes.
```
Comment (orta) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: 👍 good job

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+title:  "CocoaPods 1.6.0 Beta Begins!"
+author: dnkoutso
+categories: cocoapods releases
+---
+
+_CocoaPods 1.6.0_ release will focus on stability, performance and scalability for large projects through a longer beta cycle.
+
+<!-- more -->
+
+We are happy to announce that the first beta of 1.6.0 is out! For this release, we are revamping the build settings generation process to be faster and easier to maintain going forward and we've enhanced integration of test specs!
```
Comment (segiddins) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]:
```
In this release, we revamped the build settings generation process to be faster and easier to maintain going forward. We've also enhanced  the integration of multiple test specs in the same pod.
```

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+### Revamped Build Settings
+
+The generation of build settings (which reside within the `.xcconfig` files CocoaPods generates) has been completely re-written. The primary goal for this re-write was to clean up the code and make it scale much better for larger projects. 
+
+When contributors to CocoaPods make build setting changes it would often cause a cascading amount of performance regressions and trigger an unusual combination of edge cases. This made it very hard to test these changes and be certain that each fix works correctly for everyone.
+
+From a user point of view, this re-write should have zero effects when you compile your code. If there is anything that does not look right please file an [issue](https://github.com/CocoaPods/CocoaPods/issues/new) with us providing steps and a sample app showing the problem.
```
Comment (segiddins) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: `please file an issue, providing steps & a sample app that demonstrates the problem`

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+#### LinkedIn Performance Case Study?
+
+To demonstrate the results of this re-write we worked closely with our friends at LinkedIn, who use CocoaPods on a large project. The re-write, along with some other performance improvements, have decreased their `pod install` times down to approximately 40 seconds compared to the original time of 3 minutes! Yes it was this bad.
+
+This change, while mostly invisible, helps us guarantee that CocoaPods can scale with your app as it grows.
+
+# TODO GET PERMISSION FROM LINKEDIN TO POST THIS.
```
Comment (segiddins) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: TODO

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  # Unit Test Sources - Those do not require an app host to run. 
+  # They also require 'OCMock' dependency.
+  s.test_spec 'Tests' do |test_spec|
+	  test_spec.source_files = 'Tests/**/*.{h,m}'
+	  test_spec.dependency 'OCMock'
+  end
+
+  # SnapShot Tests Sources - Those *do* require an app host to run.
+  s.test_spec 'SnapshotTests' do |test_spec|
+   	test_spec.requires_app_host = true
```
Comment (segiddins) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: inconsistent indentation

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+	  test_spec.source_files = 'SnapshotTests/**/*.{h,m}'
+  end
+```
+
+Both the 'Tests' and 'SnapshotTests' test specs would _merge_ into a single unit test bundle target. This also had the unwanted effect of using an app host for the 'Tests' test spec without explicitly having declared by the pod author.
```
Comment (segiddins) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: `explicitly having been declared`

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+Before 1.6.0, it looks like this:
```
Comment (segiddins) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: `looked`

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+<center>
+{% breaking_image /assets/blog_img/CocoaPods-1.6.0-beta/test_specs_before.png %}
+</center>
+
+This means that pod authors would find it impossible to split their test sources and categorize them in different ways, for example using a different set of dependencies, resources or compiler options.
```
Comment (segiddins) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: `resources, or`

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+
+With 1.6.0 this will now generate the following:
```
Comment (segiddins) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: `1.6.0,`

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 13]
```diff
@@ -0,0 +1,96 @@
+---
+layout: post
+title:  "CocoaPods 1.6.0 Beta Begins!"
+author: dnkoutso
+categories: cocoapods releases
+---
+
+_CocoaPods 1.6.0_ release will focus on stability, performance and scalability for large projects through a longer beta cycle.
```
Comment (paulb777) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: The phrase at the end makes the first sentence awkward. I suggest moving `through a longer beta cycle` to another sentence.

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+
+<!-- more -->
+
+We are happy to announce that the first beta of 1.6.0 is out! For this release, we are revamping the build settings generation process to be faster and easier to maintain going forward and we've enhanced integration of test specs!
+
+### Revamped Build Settings
+
+The generation of build settings (which reside within the `.xcconfig` files CocoaPods generates) has been completely re-written. The primary goal for this re-write was to clean up the code and make it scale much better for larger projects.
```
Comment (paulb777) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: remove dashes in re-writ*

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+
+The generation of build settings (which reside within the `.xcconfig` files CocoaPods generates) has been completely re-written. The primary goal for this re-write was to clean up the code and make it scale much better for larger projects. 
+
+When contributors to CocoaPods make build setting changes it would often cause a cascading amount of performance regressions and trigger an unusual combination of edge cases. This made it very hard to test these changes and be certain that each fix works correctly for everyone.
+
+From a user point of view, this re-write should have zero effects when you compile your code. If there is anything that does not look right please file an [issue](https://github.com/CocoaPods/CocoaPods/issues/new) with us providing steps and a sample app showing the problem.
+
+#### LinkedIn Performance Case Study?
+
+To demonstrate the results of this re-write we worked closely with our friends at LinkedIn, who use CocoaPods on a large project. The re-write, along with some other performance improvements, have decreased their `pod install` times down to approximately 40 seconds compared to the original time of 3 minutes! Yes it was this bad.
```
Comment (paulb777) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: s/compared to/from/

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+
+This change, while mostly invisible, helps us guarantee that CocoaPods can scale with your app as it grows.
```
Comment (paulb777) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: s/can/will/

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+
+# TODO GET PERMISSION FROM LINKEDIN TO POST THIS.
+
+### Enhanced Test Spec Integration
+
+Test specs are becoming a fundamental piece of CocoaPods for pod authors. The ability to specify your test sources and have them execute through `lint` lets the feature become an invaluable tool for using the `podspec` file as a source of truth for your library. With 1.6.0, specifying multiple `test_spec` entries in your podspec will no longer be merged together into a single test bundle target but instead generate a unit test bundle for each one.
```
Comment (paulb777) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: s/specifying multiple `test_spec` entries in your podspec will no longer be merged together into/multiple  `test_spec` entries will no longer generate/

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+
+Much better!
+
+We _strongly_ support the idea of having tests for your code and we aim to build the right tools for doing this simply with CocoaPods. Having your `podspec` file be the source of truth and coupled with recently released [cocoapods-generate](https://github.com/square/cocoapods-generate) plugin your developer experience can be much improved and streamlined.
+
+### Longer Beta Cycle
+
+We intentionally do not commit on specific release dates for CocoaPods. This is primarily due to the fact that CocoaPods is maintained and improved by the free time of multiple people across the world. Making such commitments would be tough to meet.
+
+The 1.6.0 release will run through a longer beta cycle and will carry the 'beta' name for sometime before it's stable. We encourage you to use the 1.6.0 beta and help us by submitting any issues you find.
```
Comment (paulb777) [inline _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: s/sometime/some time/

File: _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown (hunk 8 above)
Comment (dnkoutso) [reply _posts/2018-05-08-CocoaPods-1.6.0-beta.markdown]: hm I am not sure. It seems that it might be the way its rendered? anyway I verified again its correct
